src/coprocessor/dag/batch_executor/index_scan_executor.rs

```diff
@@ -0,0 +1,515 @@
+                ft
+            },
+            {
+                let mut ft = FieldType::new();
+                ft.as_mut_accessor().set_tp(FieldTypeTp::Double);
+                ft
+            },
+            {
+                let mut ft = FieldType::new();
+                ft.as_mut_accessor().set_tp(FieldTypeTp::LongLong);
+                ft
+            },
+        ];
+
+        // Case 1. Normal index.
+
+        // For a normal index, the PK handle is stored in the key and nothing interesting is stored
+        // in the value. So let's build corresponding KV data.
+
+        let store = {
+            let kv = data
+                .iter()
+                .map(|datums| {
+                    let index_data = datum::encode_key(datums).unwrap();
+                    let key = table::encode_index_seek_key(TABLE_ID, INDEX_ID, &index_data);
+                    let key = Key::from_raw(key.as_slice());
+                    let value = vec![];
+                    (key, Ok(value))
+                })
+                .collect();
+            FixtureStore::new(kv)
+        };
+
+        {
+            // Case 1.1. Normal index, without PK, scan total index in reverse order.
+
+            let key_ranges = vec![{
+                let mut range = KeyRange::new();
+                let start_data = datum::encode_key(&[Datum::Min]).unwrap();
+                let start_key = table::encode_index_seek_key(TABLE_ID, INDEX_ID, &start_data);
+                range.set_start(start_key);
+                let end_data = datum::encode_key(&[Datum::Max]).unwrap();
+                let end_key = table::encode_index_seek_key(TABLE_ID, INDEX_ID, &end_data);
+                range.set_end(end_key);
+                range
+            }];
+
+            let mut executor = BatchIndexScanExecutor::new(
+                store.clone(),
+                Arc::new(EvalConfig::default()),
+                vec![columns_info[0].clone(), columns_info[1].clone()],
+                key_ranges,
+                true,
+                false,
+            )
+            .unwrap();
+
+            let mut result = executor.next_batch(10);
+            assert!(result.is_drained.as_ref().unwrap());
+            assert_eq!(result.data.columns_len(), 2);
+            assert_eq!(result.data.rows_len(), 3);
+            assert!(result.data[0].is_raw());
+            result.data[0].decode(&Tz::utc(), &schema[0]).unwrap();
+            assert_eq!(
+                result.data[0].decoded().as_int_slice(),
+                &[Some(5), Some(5), Some(-5)]
+            );
+            assert!(result.data[1].is_raw());
+            result.data[1].decode(&Tz::utc(), &schema[1]).unwrap();
+            assert_eq!(
+                result.data[1].decoded().as_real_slice(),
+                &[Some(10.5), Some(5.1), Some(0.3)]
+            );
+        }
+
+        {
+            // Case 1.2. Normal index, with PK, scan index prefix.
```

Comment:
One case one test?

Reply:
They share some common things, maybe current one is enough.